webapp/app/plugins/controller.js

```diff
@@ -235,25 +231,49 @@ treeherder.controller('PluginCtrl', [
          * Get the build_id needed to cancel or retrigger from the currently
          * selected job.
          */
-        var getRequestId = function() {
+        var getBuildbotRequestId = function() {
             if ($scope.artifacts.buildapi) {
                 return $scope.artifacts.buildapi.blob.request_id;
-            } else {
-                // this is super unlikely since we'd need to have at least one of those
-                // artifacts to even create the job in treeherder.  This is just a fallback...
-                thNotify.send("Unable to get request id for retrigger/cancel", "danger", true);
-                return null;
             }
         };
 
         $scope.retriggerJob = function() {
-            thBuildApi.retriggerJob($scope.repoName, getRequestId());
+            // The logic here is somewhat complicated because we need to support
+            // two use cases the first is the case where we notify a system
+            // other then buildbot that a retrigger has been requested. The
+            // second is when we have the buildapi id and need to send a request
+            // to the self serve api (which does not listen over pulse!).
+            ThJobModel.retrigger($scope.repoName, $scope.job.id).then(function() {
+                // XXX: Remove this after 1134929 is resolved.
+                var requestId = getBuildbotRequestId();
+                if (requestId) {
+                    return thBuildApi.retriggerJob($scope.repoName, requestId);
+                }
+            }).catch(function(e){
+              // Always send a message even if we have no idea what the error
+              // is.
+              var message = "Unable to send retrigger"
+
+              // If we can figure out something from the server return that.
+              if (e && e.data && e.data.detail) {
+                  message += ': ' + e.data.detail;
+              }
+
+              thNotify.send(message, "danger", true)
+            });
         };
 
         $scope.cancelJob = function() {
-            thBuildApi.cancelJob($scope.repoName, getRequestId()).then(function() {
-                ThJobModel.cancel($scope.repoName, $scope.job.id);
-            });
+            var requestId = getBuildbotRequestId();
+            if (requestId) {
+              thBuildApi.cancelJob($scope.repoName, requestId).then(function() {
+                  ThJobModel.cancel($scope.repoName, $scope.job.id);
+              });
+            } else {
+              thNotify.send(
+                "Unable to get request id for retrigger/cancel", "danger", true
+              );
+            }
         };
 
         $scope.cancelAll = function(resultsetId) {
```

Comment:
is there any concept of the pulse cancel for a group of jobs like this for the result set?